sdk/tests/system/test_code_agents.py

```diff
@@ -95,6 +123,32 @@ async def test_streaming_unhandled_error(self, server, program):
         assert exc.value.response.status_code == 500
         assert exc.value.response.json() == "big bad server error"
 
+    async def test_lots_o_context(self, server):
+        agent = Agent.get("HelloWorld")
+        resp = await agent.program("lots_o_context")
+        assert resp.data == "12"
+
+    async def test_lots_o_context_streaming(self, server):
+        agent = Agent.get("HelloWorld")
+        events = [e async for e in agent.stream_program("lots_o_context")]
+        assert events[1:-1] == [
+            StringOutputEvent(content="1"),
+            StringOutputEvent(content="2"),
+            StartStreamContextEvent(context_id="c1"),
+            StringOutputEvent(content="3", stream_context="c1"),
+            StringOutputEvent(content="4", stream_context="c1"),
+            EndStreamContextEvent(context_id="c1"),
+            StartStreamContextEvent(context_id="c2"),
+            StringOutputEvent(content="5", stream_context="c2"),
+            StringOutputEvent(content="6", stream_context="c2"),
+            StartStreamContextEvent(context_id="c3", stream_context="c2"),
+            StringOutputEvent(content="7", stream_context="c2.c3"),
+            StringOutputEvent(content="8", stream_context="c2.c3"),
+            EndStreamContextEvent(stream_context="c2", context_id="c3"),
+            EndStreamContextEvent(context_id="c2"),
+            AgentStateEvent(state="terminated", available_actions=[]),
+        ]
```

Comment:
Threw in a few tests since this was a little too much to leave untested.

Probably could have been a unit test, but I wanted to validate the e2e scenario in case agentcontroler mutates or something like that.